### ISSUE TITLE
Add Custom Fonts + Font Stylization

### ISSUE THREAD

**Issue by TheSquall** [GlitchedMythos/se-widgets#5]
Feature Request: Add the ability for non-Google custom fonts (I think it just needs to be on the host computer). Additionally, add the ability for font stylization as allowed per the font (for instance, I use Orbitron which has "Regular", "Medium", "Bold", and "Black" styles).

**Comment by billyjbryant** [comment]
I believe what you are aiming at is to allow use of any font the browser has access to, this could be done using the [`document.fonts` API](https://developer.mozilla.org/en-US/docs/Web/API/Document/fonts).  

I was able to dig up a quick example on this and found the following on Stackoverflow: https://stackoverflow.com/a/62399430

* Returned values are verbose with weight, style, etc.
```
function listFonts() {
  let { fonts } = document;
  const it = fonts.entries();

  let arr = [];
  let done = false;

  while (!done) {
    const font = it.next();
    if (!font.done) {
      arr.push(font.value[0]);
    } else {
      done = font.done;
    }
  }

  return arr;
}
```

* Returns only the font family
```
function listFonts() {
  let { fonts } = document;
  const it = fonts.entries();

  let arr = [];
  let done = false;

  while (!done) {
    const font = it.next();
    if (!font.done) {
      arr.push(font.value[0].family);
    } else {
      done = font.done;
    }
  }

  // converted to set then arr to filter repetitive values
  return [...new Set(arr)];
}
```

**Comment by billyjbryant** [comment]
So I spent a bit of time re-thinking this and while I think we could technically have a way of finding out what fonts are installed or available to the browser, there'd be no way to pick the font from a drop-down list in the Widget Options because those options are set in the JSON for the Fields. 

The `SE_API` has the ability to set a field and detect if the widget is being seen in editor mode; however, for it to do what you want, there would need to be a div selector that is only displayed in editor mode that you'd use to choose the font and the update the field in the editor using an `onChange()` event. 

It's do-able but won't be an easy feat. Might be easier to simply specify the font-family in a text field; however, you'd need to know the exact font family for the font you want to use.

**Comment by GlitchedMythos** [comment]
Yeah. I was wondering about this over the past day or so since I had seen it again.

I think there is a "way", but it is very obtuse.

If, we can get a scrollbar in a widget, then in editor mode, we could load in all the fonts and display them as a list. The user could scroll through them and then copy pasta.

They could then hide or delete that font widget.

Very obtuse way of doing it.
